handlers/slot.go

```diff
@@ -205,14 +206,15 @@ func getAttestationsData(slot uint64, onlyFirst bool) ([]*types.BlockPageAttesta
 }
 
 func GetSlotPageData(blockSlot uint64) (*types.BlockPageData, error) {
+	latestFinalizedEpoch := services.LatestFinalizedEpoch()
 	blockPageData := types.BlockPageData{}
 	blockPageData.Mainnet = utils.Config.Chain.Config.ConfigName == "mainnet"
 	// for the first slot in an epoch the previous epoch defines the finalized state
 	err := db.ReaderDb.Get(&blockPageData, `
 		SELECT
 			blocks.epoch,
-			COALESCE(epochs.finalized, false) AS epoch_finalized,
-			COALESCE(prev_epoch.finalized, false) AS prev_epoch_finalized,
+			(COALESCE(epochs.epoch, 0) <= $3) AS epoch_finalized,
```

Comment:
**issue:** The **COALESCE** behaviour would change from **false** => **true** if the value is not found.

Reply:
Thought it through ... the COALESCE is just a precaution and should not trigger. Only if we have some bigger problem in the data - and then this is the least of the problems.